### PR TITLE
making the nav drawer logo configurable #941

### LINKED ISSUE #941: Make sidebar logo configurable
**Description**:
Currently the logo in the navigation drawer is hardcoded to be the STFC logo. For other facilities, this is not suitable.

**Acceptance criteria**:
- [ ] Navigation drawer logo is configurable

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description
- Making the navigation drawer logo configurable
- Fixing the overlapping title button in the main app bar
- Changing the colour of the cookie consent to comply with WCAG 2.1 AA contrast guild lines
- Using fit-content for the min-width instead of an arbitrary number in the index.css. This insures that there is no black bar on the side when a user zooms in
 

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [x] Review code
- [x] Check Actions build
- [x] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
closes #941 , /ral-facilities/datagateway/issues/1121

### CODE REVIEW

**Comment by joshuadkitenge** [comment]
## dark mode 
- imageUrl:"https://www.diamond.ac.uk/.resources/DiamondLightModule/webresources/img/topmenulogo.png"

Screenshot 2022-02-16 112329

## light mode  
ImageUrl : "https://www.diamond.ac.uk/.resources/DiamondLightModule/webresources/img/Diamond-logo-colour.png",
 
Screenshot 2022-02-16 112249

**Comment by joshuadkitenge** [comment]
## Title button fix 
Screenshot 2022-02-16 112843

**Comment by joshuadkitenge** [comment]
## Cookie consent

###  high contrast dark mode 
Screenshot 2022-02-16 113138

### dark mode, light mode and high contrast light mode 
Screenshot 2022-02-16 113044

File: src/navigationDrawer/navigationDrawer.component.tsx [hunk 1]
```diff
@@ -158,7 +159,32 @@ class NavigationDrawer extends Component<CombinedNavigationProps> {
...
+    let altTxt = getString(this.props.res, 'alternative-text');
+
+    if (this.props.navigationDrawerLogo) {
+      if (this.props.darkMode) {
+        if (this.props.navigationDrawerLogo.dark) {
+          navDrawerLogo = this.props.navigationDrawerLogo.dark;
+          altTxt = this.props.navigationDrawerLogo.altTxt;
+        } else {
+          navDrawerLogo = STFCLogoWhiteText;
+          altTxt = getString(this.props.res, 'alternative-text');
```
Comment (louise-davies) [inline src/navigationDrawer/navigationDrawer.component.tsx:175]: We don't need the else statements here right (or in the light mode section either)? As the default logo is already set on line 162

File: src/navigationDrawer/navigationDrawer.component.tsx [hunk 2]
```diff
@@ -158,7 +159,32 @@ class NavigationDrawer extends Component<CombinedNavigationProps> {
...
-    const imgSrc = this.props.darkMode ? STFCLogoWhiteText : STFCLogoBlueText;
+    let navDrawerLogo = this.props.darkMode
+      ? STFCLogoWhiteText
+      : STFCLogoBlueText;
+
+    let altTxt = getString(this.props.res, 'alternative-text');
+
+    if (this.props.navigationDrawerLogo) {
+      if (this.props.darkMode) {
+        if (this.props.navigationDrawerLogo.dark) {
```
Comment (louise-davies) [inline src/navigationDrawer/navigationDrawer.component.tsx:170]: If you use my suggestion of error checking the settings file to ensure both logos are defined if the overall logo config is defined, you can remove these if statements

**Review by louise-davies — APPROVED** [review]
Looks good - I'll try deploying & using this next week :)

**Pull request merged**